src/components/Matches/FileExporter/index.tsx

```diff
@@ -1,52 +1,54 @@
-// @flow
 import React from 'react';
-import T from 'prop-types';
 import File from 'components/File';
+import { SupportedExtensions } from 'components/File/FileButton';
 
-const endpoint = {
+type SupportedEndpoints = 'entry' | 'protein' | 'structure';
+const endpoint: Record<
+  SupportedEndpoints,
+  Record<SupportedEndpoints, string | undefined>
+> = {
   protein: {
     entry: 'proteinEntry',
+    protein: undefined,
     structure: 'proteinStructure',
   },
   structure: {
     entry: 'structureEntry',
     protein: 'structureProtein',
+    structure: undefined,
   },
   entry: {
+    entry: undefined,
     protein: 'entryProtein',
     structure: 'entryStructure',
   },
 };
 
-const FileExporter = (
-  {
-    description,
-    search = {},
-    count,
-    fileType,
-    primary,
-    secondary,
-    className,
-    focused = null,
-    minWidth,
-  } /*: {
-    description: {
-      main: {key:string},
-      taxonomy?: {accession: string},
-    },
-    search?: {
-      extra_fields: string,
-    },
-    count: number,
-    minWidth?: number |string,
-    fileType: string,
-    primary: string,
-    secondary: string,
-    label?: string,
-    className?: string,
-    focused?: ?string
-  } */,
-) => {
+type Props = {
+  description: InterProPartialDescription;
+  search?: {
+    extra_fields?: string;
+  };
+  count: number;
+  minWidth?: number | string;
+  fileType: SupportedExtensions;
+  primary: SupportedEndpoints;
+  secondary: SupportedEndpoints;
+  label?: string;
+  focused?: string | null;
+  className?: string;
+};
+const FileExporter = ({
+  description,
+  search = {},
+  count,
+  fileType,
+  primary,
+  secondary,
+  className,
+  focused = null,
```

Comment:
Exposing `label` (defined in props):

```suggestion
  label,
  focused = null,
```